### ISSUE TITLE
Have  access to the coverage report

### ISSUE THREAD

**Issue by arthurgustin** [360EntSecGroup-Skylar/goreporter#40]
Hello,
In my gitlab-ci pipeline, I currently have 2 jobs before building sources:
   - code_quality: that mainly use goreporter
   - test: that runs all tests and export a html report of the code coverage

My main aim is to merge those 2 jobs into 1.
For this, I would like to have access to the coverage report when I click on the coverage button of the default html template of goreporter.

For instance, this is the code I actually use in my job:

```
cd MYPROJECT
ginkgo -r -cover
for f in `find . -name "*.coverprofile"`; do cat $f >> coverage/MYPROJECT.coverprofile; done;
sed -i -e '2,${ /^mode.*atomic/d }' coverage/MYPROJECT.coverprofile
go tool cover -func=coverage/MYPROJECT.coverprofile
go tool cover -html=coverage/MYPROJECT.coverprofile -o coverage/MYPROJECT.coverprofile.html
```
It just merges all coverage report into 1 and then create the html.

**Comment by wgliang** [comment]
That will be a good job!

**Comment by brokenpeace** [comment]
I feel completely the opposite. I think they serve independent purposes and coupling is exactly we should not be doing. Id honestly vote for breaking most of this apart into sub module then hopefully independent libraries... 
I love what the intention is but I feel this needs to become more DRY and focus on Single Responsibility Principles to allow or not certain aspects and easier contribution of extensions...  My opinion this reminds me of looking a monolithic Java App

**Comment by wgliang** [comment]
Yes, you are right, it should maintain functional independence. GoReporter is currently a tool, but more precisely an integrated tool set. So all the features to respond to code quality will be its area. 

I think it is important that how can we maintain functional independence and integability. Goreporter is more like a platform, do not want it to contain all the features, but will accept all the features to analyze the code, evaluation code and help developers find code defects and problems.

So, the main question we are facing is how to add this feature.
This is my point of view. @brokenpeace

**Comment by brokenpeace** [comment]
Easy don’t sleep - Assign someone to take charge of rearchitecting PM for the direction and processes, Someone to help write up a general roadmap ^^ was a decent start but broken out. Perhaps we could setup a tally or hangout and see what resources we can depend on core developing). As from a technical aspect. I can do an in-depth review and write up a technical plan of action for this. I truly do not have enough Time to carry it alone :). I have no issues PM and planning it out but core resources and roles would be nice to establish on a hangout (or whatever).
I think if it is done the sky is the limit to the framework and what you plug in and monitor. Let me know your thoughts on that?

**Comment by brokenpeace** [comment]
Oh and to alleviate this in the future probably whoever is the best technical writer (sites the roadmap) write up a roadmap and code conduct (limit who can merge Prs). Because some of what I’ve read o this project is a lot of heavy over engineering developers who influence majorrr scope creep and limitations.. Need declaration of code :P conduct hahaha.  To eliminate these scary ideas!@

First step is to organize out who we have ad who can commit, 2nd DRYup this code and rip out the components and heavy refactoring. around the base with modularity and independence (85%+ test coverage) and a test with any. new feature etc... tons or ideas to plan it but #1

**Comment by arthurgustin** [comment]
Sounds like a good idea. +1 for tests. I'd be glad to participate in further discussions !

**Comment by brokenpeace** [comment]
Who manages the project? is it ok if I get project mgmnt accesss? to jot some stories etc. Arthur can you gather the actives and see who is interested in helping? I am avail to meet wed and thursor next week.

**Comment by arthurgustin** [comment]
@wgliang is the creator of the project, you will have to see with him for this kind of questions. Btw I have the feeling this is an opensourced company project, we don't have our word to say about how the project should be driven.

**Comment by wgliang** [comment]
No,no,no. You do not need to worry about this, as long as it is conducive to the development of the project, any changes I will consider accept.

**Comment by wgliang** [comment]
If you have good suggestions or ideas, as far as possible directly to mention. You can mention PRs, Issues or other ways you like. @arthurgustin @brokenpeace
